### PR TITLE
fix(has-many-dissociator): dissociate relationship when related record is deleted

### PR DESCRIPTION
**Pull request by olesyakorovina**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [x] Create automatic tests
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

**Review by VincentMolinie — CHANGES_REQUESTED** [review]
Works perfectly fine BUT it's missing a test 😉

File: test/tests/services/has-many-dissociator.test.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+          owners: {
+            type: ['ObjectId'],
+            ref: 'LumberJack',
+          },
+        });
+
+        LumberJackModel = mongoose.model('LumberJack', LumberJackSchema);
+        TreeModel = mongoose.model('Tree', TreeSchema);
+
+        return Promise.all([LumberJackModel.remove({}), TreeModel.remove({})]);
```
Comment (VincentMolinie) [inline test/tests/services/has-many-dissociator.test.js:66]: [MISC] You are doing this in the `beforeAll` and in the `beforeEach` 🤔 ? Why ?

File: test/tests/services/has-many-dissociator.test.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    loadFixture(LumberJackModel, [
+      {
+        _id: '41224d776a326fb40f000001',
+        name: 'Kaladin',
+      },
+      {
+        _id: '41224d776a326fb40f000002',
+        name: 'Adolin Kholin',
+      },
+    ]), loadFixture(TreeModel, [
```
Comment (VincentMolinie) [inline test/tests/services/has-many-dissociator.test.js:84]: [Code Style] A new line before the loadFixtures here please 🙏

File: test/tests/services/has-many-dissociator.test.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
+import mongoose from 'mongoose';
+import loadFixture from 'mongoose-fixture-loader';
+import Interface from 'forest-express';
+import mongooseConnect from '../../utils/mongoose-connect';
+import HasManyDissociator from '../../../src/services/has-many-dissociator';
+
+describe('service > has-many-dissociator', () => {
```
Comment (VincentMolinie) [inline test/tests/services/has-many-dissociator.test.js:7]: [Wording] `Service > hasManyDissociator` looks better. And I think we already formatted the other like that

**Review by VincentMolinie — CHANGES_REQUESTED** [review]
Good job for the tests 👏 ! Almost good 😄

File: test/tests/services/has-many-dissociator.test.js (hunk 3 above)
Comment (olesyakorovina) [reply test/tests/services/has-many-dissociator.test.js:7]: for all other services, the dashed version is used in the tests, so I used it to align with others

File: test/tests/services/has-many-dissociator.test.js (hunk 1 above)
Comment (olesyakorovina) [reply test/tests/services/has-many-dissociator.test.js:66]: this was a copy-paste from `resource-remover.test.js` but it's true that it doesn't make much sense, I will keep it only in the `beforeEach`

File: test/tests/services/has-many-dissociator.test.js (hunk 3 above)
Comment (VincentMolinie) [reply test/tests/services/has-many-dissociator.test.js:7]: Okey my bad

**Review by VincentMolinie — APPROVED** [review]

**Pull request merged**